sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,131 @@
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+import DVR
+
+public class SubscriptionIDFilter : Filter {
+    
+    public override init() {
+        super.init()
+        let standardHeaderScrubbing: [String : FilterBehavior] = [
+            "location": .closure(scrubSubscriptionIDClosure),
+           "operation-location": .remove,
+           "azure-asyncoperation": .remove,
+           "www-authenticate": .remove,
+           "access_token": .remove,
+            "authorization": .remove,
+            "client-request-id" : .remove,
+            "retry-after" : .remove,
+            "x-ms-client-request-id": .remove,
+            "x-ms-correlation-request-id": .remove,
+            "x-ms-ratelimit-remaining-subscription-reads": .remove,
+            "x-ms-request-id":.remove,
+            "x-ms-routing-request-id": .remove,
+            "x-ms-gateway-service-instanceid": .remove,
+            "x-ms-ratelimit-remaining-tenant-reads": .remove,
+            "x-ms-served-by": .remove,
+            "x-ms-authorization-auxiliary": .remove
+       ]
+        filterHeaders = standardHeaderScrubbing
+        beforeRecordRequest = scrubSubscriptionIDs
+        beforeRecordResponse = scrubSubscriptionIDs
+    }
+    
+    static let subscriptionIDRegex = try! NSRegularExpression(pattern: "(/(subscriptions))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: .caseInsensitive)
+    static let graphSubscriptionIDRegex = try! NSRegularExpression(pattern: "(https://(graph.windows.net))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: .caseInsensitive)
+    
+    static var subcriptionIDReplacement = "00000000-0000-0000-0000-000000000000"
+    
+    /// Wraps `ScrubSubscriptionId` in a function to be passed as closure within initializer
+    func scrubSubscriptionIDClosure(key: String, val: String?) -> (String?) {
+        guard let val = val else { return nil }
+        return scrubSubscriptionId(from: val)
+    }
```

Comment:
Originally I did not want to write this but since `scrubSubscriptionId` is not  static, I couldn't see another way to get the function inside the closure for "location" since it didn't have a matching parameter and return type. What's your thoughts on this @tjprescott ?

Reply:
I think it's fine since that's the signature for the closure enum